docs/topics/OAuth2.md

```diff
@@ -56,68 +43,232 @@ Scopes provide access to certain resources of a user's account. Your API client
+>Unlike the rest of the scopes, `guilds.join` requires you to have a bot account linked to your application and can only be used to join users to guilds which your bot services.
 
-## Bots
+## Authorization Code Grant
+
+The authorization code grant is the "fullest" and most secure of the OAuth2 flavors. It allows the authorization server to act as an intermediary between the client and the resource owner, so the resource owner's credentials are never shared directly with the client. This is what most developers will recognize as "standard OAuth2" and involves retrieving an access code and exchanging it for a user's access token.
+
+###### Authorization Code URL Example
+
+```
+https://discordapp.com/oauth2/authorize?response_type=code&client_id=A&scope=B&redirect_uri=C
+```
+
+`client_id` is your application's `client_id`. `scope` is a list of [OAuth2 scopes](#DOCS_OAUTH2/shared-resources-oauth2-scopes) separated by '+'. `redirect_uri` is whatever URL you registered when creating your application, url-encoded. In practice, the entire URL will look something like this:
+
+```
+https://discordapp.com/oauth2/authorize?response_type=code&client_id=157730590492196864&scope=identify+guilds.join&redirect_uri=https%3A%2F%2Fnicememe.website
+```
+
+When someone navigates to this URL, they will be prompted to authorize your application for the requested scopes. On acceptance, they will be redirected to your `redirect_uri`, which will contain an additional querystring parameter, `code`. This should be exchanged for the user's access token by making a `POST` request to the [token URL](#DOCS_OAUTH2/shared-resources-oauth2-urls) with the following parameters:
+
+- `client_id` - your application's client id
+- `client_secret` - your application's client secret
+- `grant_type` - must be set to `authorization_code`
+- `code` - the code from the querystring
+- `redirect_uri` - your `redirect_uri`
+
+###### Access Token Exchange Example
+
+```python
+def exchangeCode(code):
+  data = {
+    'client_id': '332269999912132097',
+    'client_secret': '456265548123452097',
+    'grant_type': 'authorization_code',
+    'code': code,
+    'redirect_uri': 'https://nicememe.website'
+  }
+  headers = {
+    'Content-Type': 'application/x-www-form-urlencoded'
+  }
+  return requests.post('https://discordapp.com/api/oauth2/token', data, headers).json()
+```
+
+You can also pass your `client_id` and `client_secret` as an authentication header of type `Basic` and a value of the base64 encoded string formatted as "client\_id:client_secret":
+
+###### Access Token Exchange Authentication Header Example
+
+```python
+import base64
+def exchangeCode(code):
+  data = {
+    'grant_type': 'authorization_code',
+    'code': code,
+    'redirect_uri': 'https://nicememe.website'
+  }
+  headers = {
+    'Content-Type': 'application/x-www-form-urlencoded'
+    'Authorization': 'Basic ' + base64.b64encode(b'332269999912132097:456265548123452097')
```

Comment:
u can also pass this via `requests`'s `auth` kwarg, but maybe this is better so end user can clearly see what is happening.